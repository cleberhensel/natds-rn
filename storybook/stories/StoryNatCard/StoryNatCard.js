import React from 'react';
import {
  ScrollView,
  SafeAreaView,
  Dimensions,
} from 'react-native';
import { NatContainer, NatCard } from '../../../lib';

const styles = {
  defaultScreen: {
    backgroundColor: '#FAFAFA',
    marginTop: 0,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatCard = () => (
  <SafeAreaView>
    <NatContainer style={styles.defaultScreen}>
      <ScrollView>
        <NatCard
          title="Card Test"
          subtitle="13 hours ago"
          resume="Esse Ã© um card da Natura"
          iconURI="https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png"
          imageURI="https://i.picsum.photos/id/1016/328/184.jpg"
          onPress={() => null}
        />
      </ScrollView>
    </NatContainer>
  </SafeAreaView>

);
