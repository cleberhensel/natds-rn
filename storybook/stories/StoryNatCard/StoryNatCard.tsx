import React from 'react';
import { ScrollView, SafeAreaView, Dimensions } from 'react-native';
import { NatContainer, NatCard, NatSpace } from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatCard = () => (
  <SafeAreaView style={styles.defaultScreen}>
    <NatContainer style={{}}>
      <ScrollView>
        <NatCard
          title="Card Test"
          subtitle="13 hours ago"
          resume="Esse é um card da Natura"
          iconURI="https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png"
          imageURI="https://i.picsum.photos/id/1016/328/184.jpg"
          onPress={() => null}
        />
        <NatSpace />
        <NatCard
          title="Card Test"
          subtitle="13 hours ago"
          resume="Esse é um card da Natura"
          iconURI="https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png"
          imageURI="https://i.picsum.photos/id/1016/528/184.jpg"
          onPress={() => null}
        />
        <NatSpace />
        <NatCard
          title="Card Test"
          subtitle="13 hours ago"
          resume="Esse é um card da Natura"
          iconURI="https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png"
          imageURI="https://i.picsum.photos/id/1016/728/184.jpg"
          onPress={() => null}
        />
        <NatSpace />
        <NatCard
          title="Card Test"
          subtitle="13 hours ago"
          resume="Esse é um card da Natura"
          iconURI="https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png"
          imageURI="https://i.picsum.photos/id/1016/200/184.jpg"
          onPress={() => null}
        />
        <NatSpace />
      </ScrollView>
    </NatContainer>
  </SafeAreaView>
);
