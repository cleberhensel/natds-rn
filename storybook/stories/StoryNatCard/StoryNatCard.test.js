import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { StoryNatCard } from './StoryNatCard';

describe('StoryNatCard component', () => {
  it('Should render StoryNatCard component ', () => {
    const button = renderer.create(
      <StoryNatCard />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
