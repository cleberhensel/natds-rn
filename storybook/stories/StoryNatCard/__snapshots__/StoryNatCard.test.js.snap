// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatCard component Should render StoryNatCard component  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
<<<<<<< HEAD
=======
<<<<<<< HEAD
=======
<<<<<<< HEAD
=======
<<<<<<< HEAD
=======
<<<<<<< HEAD
>>>>>>> origin/update/react
>>>>>>> origin/update/react
>>>>>>> origin/update/react
>>>>>>> origin/update/react
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "bottom": 0,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "right": 0,
          "top": 0,
        },
        Object {
<<<<<<< HEAD
          "backgroundColor": "#FAFAFA",
          "marginTop": 0,
          "width": 750,
=======
<<<<<<< HEAD
          "backgroundColor": "#FAFAFA",
          "marginTop": 0,
          "width": 750,
=======
          "backgroundColor": "#F5F5F5",
>>>>>>> origin/update/react
>>>>>>> origin/update/react
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
<<<<<<< HEAD
=======
<<<<<<< HEAD
=======
        <RCTScrollView>
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
<<<<<<< HEAD
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "marginTop": 0,
                  "opacity": 1,
                }
=======
<<<<<<< HEAD
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "marginTop": 0,
                  "opacity": 1,
                }
=======
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "marginTop": 0,
                  "opacity": 1,
                }
=======
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "marginTop": 0,
            "opacity": 1,
          }
        }
      >
>>>>>>> origin/update/react
>>>>>>> origin/update/react
        <View
          style={
            Object {
              "borderRadius": 4,
              "elevation": 1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 4,
            }
          }
        >
          <View
<<<<<<< HEAD
=======
<<<<<<< HEAD
>>>>>>> origin/update/react
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
<<<<<<< HEAD
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "marginBottom": 0,
                "opacity": 1,
              }
            }
          >
            <View
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "width": "100%",
=======
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "marginBottom": 0,
                "opacity": 1,
              }
            }
          >
            <View
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "width": "100%",
=======
            margin={24}
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <Text
              color="#333333"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFF",
                    "color": "#333333",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "lineHeight": 30,
                  },
                ]
              }
              type="h6"
            >
              Card Test
            </Text>
          </View>
          <View
            margin={24}
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <Text
              color="#333333"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFF",
                    "color": "#333333",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "lineHeight": 18,
>>>>>>> origin/update/react
>>>>>>> origin/update/react
                  },
                ]
              }
              type="card"
            >
<<<<<<< HEAD
=======
<<<<<<< HEAD
>>>>>>> origin/update/react
              <Image
                source={
                  Object {
                    "uri": "https://i.picsum.photos/id/1016/328/184.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 0,
                      "height": 184,
                      "width": "100%",
                    },
                  ]
                }
                type="card"
              />
            </View>
            <View
<<<<<<< HEAD
              style={
                Array [
                  Object {
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
=======
              style={
                Array [
                  Object {
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
=======
              13 hours ago
            </Text>
          </View>
          <View
            margin="dense"
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": "dense",
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <Text
              color="#333333"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFF",
                    "color": "#333333",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 21,
>>>>>>> origin/update/react
>>>>>>> origin/update/react
                  },
                ]
>>>>>>> origin/update/react
>>>>>>> origin/update/react
>>>>>>> origin/update/react
              }
            >
              <View
<<<<<<< HEAD
=======
<<<<<<< HEAD
>>>>>>> origin/update/react
                margin={4}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "marginBottom": 4,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <Text
                  color="#333333"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFF",
                        "color": "#333333",
                        "fontSize": 20,
                        "fontWeight": "500",
                        "lineHeight": 30,
                      },
                    ]
                  }
                  type="h6"
                >
                  Card Test
                </Text>
              </View>
              <View
                margin={16}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "marginBottom": 16,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
<<<<<<< HEAD
                    },
                  ]
                }
              >
=======
=======
                style={
                  Array [
                    Object {
                      "width": "100%",
                    },
                  ]
                }
                type="card"
              >
                <Image
                  source={
                    Object {
                      "uri": "https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "borderRadius": 0,
                        "height": 184,
                        "width": "100%",
                      },
                    ]
                  }
                  type="card"
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
>>>>>>> origin/update/react
                    },
                  ]
                }
              >
<<<<<<< HEAD
>>>>>>> origin/update/react
                <Text
                  color="#777777"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFF",
                        "color": "#777777",
                        "fontSize": 12,
                        "fontWeight": "400",
                        "lineHeight": 18,
                      },
                    ]
                  }
                  type="caption"
                >
                  13 hours ago
                </Text>
              </View>
              <View
                margin={24}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "marginBottom": 24,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <Text
                  color="#333333"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFF",
                        "color": "#333333",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "lineHeight": 21,
                      },
                    ]
                  }
                  type="body2"
                >
                  Esse é um card da Natura
                </Text>
<<<<<<< HEAD
=======
=======
                <View
                  margin={24}
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "marginBottom": 24,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFF",
                          "color": "#333333",
                          "fontSize": 20,
                          "fontWeight": "500",
                          "lineHeight": 30,
                        },
                      ]
                    }
                    type="h6"
                  >
                    Card Test
                  </Text>
                </View>
                <View
                  margin={24}
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "marginBottom": 24,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFF",
                          "color": "#333333",
                          "fontSize": 12,
                          "fontWeight": "400",
                          "lineHeight": 18,
                        },
                      ]
                    }
                    type="caption"
                  >
                    13 hours ago
                  </Text>
                </View>
                <View
                  margin="dense"
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "marginBottom": "dense",
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFF",
                          "color": "#333333",
                          "fontSize": 14,
                          "fontWeight": "400",
                          "lineHeight": 21,
                        },
                      ]
                    }
                    type="body2"
                  >
                    Esse é um card da Natura
                  </Text>
                </View>
>>>>>>> origin/update/react
>>>>>>> origin/update/react
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text>
           You can apply margin-top by sending “margin“ prop, with values: none, dense or normal. 
        </Text>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
